apps/web/src/pages/workspaces/index.tsx

```diff
@@ -7,13 +7,19 @@ export default function WorkspacesPage() {
   const router = useRouter()
   const [{ isLoading, data }] = useWorkspaces()
   const signOut = useSignout()
+  const session = useSession()
 
   useEffect(() => {
-    if (isLoading) {
+    if (isLoading || session.isLoading) {
       return
     }
 
-    const workspace = data[0]
+    const workspace =
+      data.find(
+        (workspace) => workspace.id === session.data?.lastVisitedWorkspaceId
+      ) ??
+      data.find((workspace) => workspace.ownerId === session.data?.id) ??
+      data[0]
```

Comment:
Probably not something for this PR, but it's kind of annoying that we have to have very similar code here and in the session redirect.